#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_xmms-autotools.dpatch by Matthew W. S. Bell <matthew@bells23.org.uk>
##
## All lines beginning with ## DP:' are a description of the patch.
## DP: Fix the configuration of the XMMS plugin using autotools
## DP: This patch has already been applied, and autoreconf called.

@DPATCH@
diff -urNad faad2.tmp~/configure.in faad2.tmp/configure.in
--- faad2.tmp~/configure.in	2007-10-15 00:41:42.000000000 +0100
+++ faad2.tmp/configure.in	2007-10-15 00:43:05.000000000 +0100
@@ -16,9 +16,9 @@
 
 dnl Checks for programs.
 AC_PROG_CC
-AC_PROG_CPP
-dnl disable for mpeg4ip plugin
-dnl AC_PROG_CXX
+dnl AC_PROG_CPP
+dnl enable for mpeg4ip plugin, rather than above
+AC_PROG_CXX
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
@@ -129,10 +129,7 @@
 fi
 
 if test x$WITHXMMS = xyes; then
-  AC_CHECK_PROGS(XMMS_CONFIG, xmms-config,"not_found")
-  if test "$XMMS_CONFIG" = "not_found"; then
-    AC_MSG_ERROR("*** xmms-config not found - xmms plugin can't be build")
-  fi
+  AM_PATH_XMMS()
   AC_CHECK_HEADER(pthread.h,,
            AC_MSG_ERROR(*** pthread headers support not installed or not found))
   AC_CHECK_HEADER(id3.h,,
diff -urNad faad2.tmp~/plugins/xmms/src/Makefile.am faad2.tmp/plugins/xmms/src/Makefile.am
--- faad2.tmp~/plugins/xmms/src/Makefile.am	2007-10-15 00:42:18.000000000 +0100
+++ faad2.tmp/plugins/xmms/src/Makefile.am	2007-10-15 00:43:32.000000000 +0100
@@ -1,6 +1,7 @@
-local_CFLAGS=`$(XMMS_CONFIG) --cflags` -Wall
-local_LDFLAGS=`$(XMMS_CONFIG) --libs`
-libdir = `$(XMMS_CONFIG) --input-plugin-dir`
+local_CFLAGS=$(XMMS_CFLAGS) -Wall
+local_LDFLAGS=$(XMMS_LIBS)
+libdir = $(XMMS_INPUT_PLUGIN_DIR)
+  
 lib_LTLIBRARIES = libmp4.la
 
 libmp4_la_CFLAGS = $(local_CFLAGS) -Wall \
